compiler_opt/rl/log_reader_test.py

```diff
@@ -158,6 +181,83 @@ def test_pickling(self):
     for i in range(len(fv1)):
       self.assertEqual(fv1[i], o[i])
 
+  def test_seq_example_conversion(self):
+    logfile = self.create_tempfile()
+    create_example(logfile, nr_contexts=2)
+    seq_examples = log_reader.read_log_as_sequence_examples(logfile)
+    self.assertIn('context_nr_0', seq_examples)
+    self.assertIn('context_nr_1', seq_examples)
+    self.assertEqual(
+        seq_examples['context_nr_1'].feature_lists.feature_list['tensor_name1']
+        .feature[0].int64_list.value, [12, 13, 14])
+    # each context has 2 observations. The reward is scalar, the
+    # 2 features' shapes are given in `create_example` above.
+    expected_ctx_0 = text_format.Parse(
+        """
+feature_lists {
+  feature_list {
+    key: "reward"
+    value {
+      feature {
+        float_list {
+          value: 1.2000000476837158
+        }
+      }
+      feature {
+        float_list {
+          value: 2.200000047683716
+        }
+      }
+    }
+  }
+  feature_list {
+    key: "tensor_name1"
+    value {
+      feature {
+        int64_list {
+          value: 1
+          value: 2
+          value: 3
+        }
+      }
+      feature {
+        int64_list {
+          value: 2
+          value: 3
+          value: 4
+        }
+      }
+    }
+  }
+  feature_list {
+    key: "tensor_name2"
+    value {
+      feature {
+        float_list {
+          value: 0.10000000149011612
+          value: 0.20000000298023224
+          value: 0.30000001192092896
+          value: 0.4000000059604645
+          value: 0.5
+          value: 0.6000000238418579
+        }
+      }
+      feature {
+        float_list {
+          value: 1.100000023841858
+          value: 1.2000000476837158
+          value: 1.2999999523162842
+          value: 1.399999976158142
+          value: 1.5
+          value: 1.600000023841858
+        }
+      }
+    }
+  }
+}
+""", tf.train.SequenceExample())
+    self.assertEqual(expected_ctx_0, seq_examples['context_nr_0'])
```

Comment:
use `self.assertProtoEquals` which normalizes floating point numbers when comparing.

https://www.tensorflow.org/api_docs/python/tf/test/TestCase#assertProtoEquals

you will need to subclass from tf.test.TestCase and below use tf.test.main() instead of using absltest.

Reply:
done